src/script/bitcoinconsensus.cpp

```diff
@@ -83,6 +84,14 @@ static int verify_script(const unsigned char *scriptPubKey, unsigned int scriptP
     try {
         TxInputStream stream(PROTOCOL_VERSION, txTo, txToLen);
         CTransaction tx(deserialize, stream);
+        std::vector<CTxOut> spent_outputs;
+        if (spentOutputs != nullptr) {
+            TxInputStream spent_outputs_stream(PROTOCOL_VERSION, spentOutputs, spentOutputsLen);
+            spent_outputs_stream >> spent_outputs;
```

Comment:
Might be nicer to let the caller pass in an array rather than have to serialise the vector?

Reply:
Correct me if I'm wrong but I thought it would have to be a vector and that can't be exposed as part of the C API

Reply:
It could take in a pointer to an array of {pointer to utxo, length} structs or so, as well as the number of inputs. That's not particularly convenient to use, but neither is the current interface (as it probably requires serialization code on the caller side).